src/report/ReportPoller.ts

```diff
@@ -104,18 +110,29 @@ export class ReportPoller {
             });
         }
 
-        /*
-         * This API endpoint returns an opaque `next_token` number that we
-         * need to give back to subsequent requests for pagination, so here we
-         * save it in account data
-         */
-        if (response.next_token !== undefined) {
-            this.from = response.next_token;
-            try {
-                await this.mjolnir.client.setAccountData(REPORT_POLL_EVENT_TYPE, { from: response.next_token });
-            } catch (ex) {
-                await this.mjolnir.logMessage(LogLevel.ERROR, "getAbuseReports", `failed to update progress: ${ex}`);
-            }
+        let from;
+        if (this.from === null) {
```

Comment:
You're right that we want to do something to avoid displaying tens of thousands of old reports the first time we start the Report Poller on the server.

I believe that we should rather:

* Upon startup of the `ReportPoller`.
   1. Load `from` from Mjölnir's account data.
   2. If there is no `from`
      a. Fetch a cutout date from the configuration, or some default date if not specified (say August 1st 2022).
      b. Start background enumerating all reports until the cutout date, without displaying them.
      c. Use this to initialize `from`. This may also give us a first few abuse reports that will need to be returned the first time we call `getAbuseReports`.
      d. Store this value of `from` in Mjölnir's account data.
* Whenever we call `getAbuseReports`.
   1. Wait until background enumeration is complete before returning any new reports.
   2. By definition, we have a `from` (which may be `null` if the server has never encountered any abuse report).
   3. Proceed as usual.
   4. Whenever we update `from`, store it in Mjölnir's account data.

What do you think?